backends/rapidpro/backend.go

```diff
@@ -58,6 +59,51 @@ func (b *backend) GetContact(ctx context.Context, c courier.Channel, urn urns.UR
 	return contactForURN(ctx, b, dbChannel.OrgID_, dbChannel, urn, auth, name)
 }
 
+// AddURNtoContact adds a URN to the passed in contact
+func (b *backend) AddURNtoContact(context context.Context, channel Channel, contact Contact, urn urns.URN) (urns.URN, error) {
+	tx, err := b.db.BeginTxx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	dbChannel := c.(*DBChannel)
+	_, err = contactURNForURN(tx, dbChannel.OrgID_, dbChannel.ID(), contact.ID(), urn, "")
+	if err != nil {
+		return nil, err
+	}
+	return urn, nil
+}
+
+// RemoveURNFromcontact removes a URN from the passed in contact
+func (b *backend) RemoveURNfromContact(context context.Context, channel Channel, contact Contact, urn urns.URN) (urns.URN, error) {
```

Comment:
This seems a little overcomplicated.. you should be able to do this in a single query, something like:

`UPDATE contacts_contacturn SET contact_id = NULL where contact_id = $1 AND identity = $2`